sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py

```diff
@@ -464,6 +522,75 @@ def get_current_state_bounded(self):
     def get_robot_state_bounded(self):
         return self._move_group_commander._g.get_current_state_bounded()
 
+    @staticmethod
+    def _bound_joint(joint_value, joint_limits):
+        """
+        Forces a joint value to be within the given joint limits
+        @joint_value - Value of the joint to be bounded
+        @joint_limits - Tuple with the lower limit and upper limit of the joint
+        @return - Joint value within the given joints
+        """
+        if joint_value < joint_limits[0]:
+            joint_value = joint_limits[0]
+        elif joint_value > joint_limits[1]:
+            joint_value = joint_limits[1]
+        return joint_value
+
+    def _bound_state(self, joint_states):
+        """
+        Bounds the joint states within the limits of the joints
+        @param joint_states - It can be of type dict or RobotState. This param should contain the joints to be bounded
+        within their limits
+        @return - The joint states updated with the joint poisitions bounded
+        """
+        if type(joint_states) == dict:
+            for joint in joint_states:
+                joint_states[joint] = self._bound_joint(joint_states[joint],
+                                                        self._joint_limits[joint])
+
+        elif type(joint_states) == RobotState:
+            for i in range(0, len(joint_states.joint_state.name)):
+                joint_states.joint_state.position[i] = \
+                    self._bound_joint(joint_states.joint_state.position[i],
+                                      self._joint_limits[joint_states.joint_state.name[i]])
+
+        return joint_states
+
+    def get_current_set_points(self):
+        """
+        Reads from the set points
+        @return - Dictionary which contains the set points of the joints that belong to the move group
+        """
+        with self._set_points_lock:
+            raw_set_points = copy.deepcopy(self._set_points)
+
+        current_state = self.get_current_state()
+        set_points = {}
+
+        for joint in raw_set_points:
+            if "J0" in joint:
+                # Get j1 j2 ratio from current state
+                underactuation_ratio = current_state[f"{joint[:-2]}J1"]/current_state[f"{joint[:-2]}J2"]
+                ratio_part = raw_set_points[joint]/(underactuation_ratio+1)
+                set_point_j1 = underactuation_ratio*ratio_part
+                set_point_j2 = ratio_part
+                set_points.update({f"{joint[:-2]}J1": set_point_j1})
+                set_points.update({f"{joint[:-2]}J2": set_point_j2})
+            elif joint[5:] not in ["J1", "J2"] or joint[3:-2] in ["WR", "TH"]:
```

Comment:
Would be good to not use the hardcoded indexes I think, maybe use any() or give a comment on how the joint string is structured?

Reply:
I would consider using a regex to match the joints conditionally